bin2c/bin2c.c

```diff
@@ -44,19 +47,34 @@
     }
 #endif
 
+	unsigned char * fprintList(FILE *outputFile, unsigned char *bufptr, unsigned char len, const char* formatStr)
+	{
+		for (unsigned char j = 0; j < len; j++)
+		{
+			fprintf(outputFile, formatStr, *(bufptr++));
+			fprintf(outputFile, ",");
+		}
+		fprintf(outputFile, "\n");
+		return bufptr;
+	}
+
 int main(int argc, char *argv[])
 {
     FILE          *inputFile = NULL;
     FILE          *outputFile = NULL;
     void          *buffer;
-    unsigned int  fileLen, i, j, major, minor;
+    unsigned int  fileLen, i, major, minor;
     unsigned char *bufptr;
+	unsigned char useHex = 0;
 
-    if(argc != 4) {
-        printf("%s %s %s\n", "Usage:", argv[0], "<binary> <destination file> <array name>");
+    if(argc < 4 || argc > 5) {
+        printf("%s %s %s\n", "Usage:", argv[0], "<binary> <destination file> <array name> [options]");
+		printf("%s\n", "options:\n\t-h output numbers in hex format");
         return 1;
     }
 
+	useHex = strstr(argv[4], "-h") != NULL;
```

Comment:
We should be properly parsing out argv here, making the options position independent.
For example, bin2c -h <in> <out> should still be a valid invocation.
The above order would actually be the preferred position.